File: test/org/webfabric/WebServer.java
package org.webfabric;

import org.mortbay.jetty.webapp.WebAppContext;
import org.mortbay.jetty.nio.SelectChannelConnector;
import org.mortbay.jetty.Connector;
import org.mortbay.jetty.Server;

public class WebServer {
    private final int port;
    private final Server server;

    public WebServer(int portNumber) {
        this.port = portNumber;
        this.server = new Server();
        server.addConnector(CreateConnector());
        server.setHandler(CreateAppContext());
        server.setStopAtShutdown(true);
    }

    public static void main(String[] args) throws Exception {
        WebServer webServer = new WebServer(8080);
        webServer.start();
    }

    private WebAppContext CreateAppContext() {
        WebAppContext appContext = new WebAppContext();
        appContext.setContextPath("/");
        appContext.setWar("./web");
        return appContext;
    }

    private Connector CreateConnector() {
        Connector connector = new SelectChannelConnector();
        connector.setPort(port);
        return connector;
    }

    public void start() throws Exception {
        server.start();
    }
}

File: test/org/webfabric/jetty/WebServer.scala
package org.webfabric.web

import mortbay.jetty.nio.SelectChannelConnector
import mortbay.jetty.webapp.WebAppContext
import mortbay.jetty.{Connector, Server}

class WebServer(port: Int) {
  val server = new Server()
  createConnector()
  createAppContext()
  server.setStopAtShutdown(true)

  def createAppContext() {
    val appContext = new WebAppContext()
    appContext.setContextPath("/")
    appContext.setWar("./web")
    server.setHandler(appContext)
  }

  def createConnector() {
    val connector = new SelectChannelConnector()
    connector.setPort(port)
    server.addConnector(connector)
  }

  def start = {
    server.start()
  }
}

